provider/internal/java/dependency.go

```diff
@@ -241,6 +244,37 @@ func (p *javaServiceClient) GetDependenciesDAG(ctx context.Context) (map[uri.URI
 	return m, nil
 }
 
+// extractSubmoduleTrees creates an array of lines for each submodule tree found in the mvn dependency:tree output
+func extractSubmoduleTrees(lines []string) [][]string {
+	submoduleTrees := [][]string{}
+
+	beginRegex := regexp.MustCompile(`maven-dependency-plugin:[\d\.]+:tree`)
+	endRegex := regexp.MustCompile(`\[INFO\] -*$`)
+
+	submod := 0
+	gather, skipmod := false, true
+	for _, line := range lines {
+		if beginRegex.Find([]byte(line)) != nil {
+			gather = true
+			submoduleTrees = append(submoduleTrees, []string{})
+		} else if gather {
+			if endRegex.Find([]byte(line)) != nil {
```

Comment:
nit: use continue instead of else blocks 

It will make the for loop more clear.